packages/story-editor/src/components/library/panes/media/local/mediaPane.js

```diff
@@ -350,7 +360,23 @@ function MediaPane(props) {
                 )}
               </SearchCount>
             )}
-            {!isSearching && (
+            {!isSearching && enableHotlinking && (
+              <ButtonsWrapper>
+                <LinkInsertion />
+                <Tooltip title={__('Upload', 'web-stories')}>
+                  <Button
+                    variant={BUTTON_VARIANTS.SQUARE}
+                    type={BUTTON_TYPES.SECONDARY}
+                    size={BUTTON_SIZES.SMALL}
+                    onClick={openMediaPicker}
+                    aria-label={__('Upload', 'web-stories')}
+                  >
+                    <Icons.ArrowCloud />
+                  </Button>
+                </Tooltip>
+              </ButtonsWrapper>
+            )}
+            {!isSearching && !enableHotlinking && (
```

Comment:
Are you disabling upload if hotlinking is enabled? 

Let's change to match this https://github.com/google/web-stories-wp/pull/8697/files#diff-0f37d63c9ac90f48c0495fe965101310fb160575fca357e7264611315773145bR349

Reply:
No, not disabling upload, it just changes the button here.